scheduler/filter/constraint.go

```diff
@@ -26,7 +27,7 @@ func (f *ConstraintFilter) Filter(config *dockerclient.ContainerConfig, nodes []
 			switch constraint.key {
 			case "node":
 				// "node" label is a special case pinning a container to a specific node.
-				if constraint.Match(node.ID, node.Name) {
+				if constraint.Match(node.ID, node.Name, node.Addr, strings.Split(node.Addr, ":")[0]) {
```

Comment:
if `node.Addr` does not contain `:`, this would break.
could you please do some checking before using it?